hw13-ioc-container/src/test/java/ru/otus/AppTest.java

```diff
@@ -0,0 +1,123 @@
+package ru.otus;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import java.io.PrintStream;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Scanner;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import ru.otus.appcontainer.AppComponentsContainerImpl;
+import ru.otus.appcontainer.ContextException;
+import ru.otus.appcontainer.api.AppComponent;
+import ru.otus.appcontainer.api.AppComponentsContainerConfig;
+import ru.otus.config.AppConfig1;
+import ru.otus.services.*;
+
+class AppTest {
+
+    @DisplayName("Из контекста тремя способами должен корректно доставаться компонент с проставленными полями")
+    @ParameterizedTest(name = "Достаем по: {0}")
+    @CsvSource(
+            value = {
+                "GameProcessor, ru.otus.services.GameProcessor",
+                "GameProcessorImpl, ru.otus.services.GameProcessor",
+                "gameProcessor, ru.otus.services.GameProcessor",
+                "IOService, ru.otus.services.IOService",
+                "IOServiceStreams, ru.otus.services.IOService",
+                "ioService, ru.otus.services.IOService",
+                "PlayerService, ru.otus.services.PlayerService",
+                "PlayerServiceImpl, ru.otus.services.PlayerService",
+                "playerService, ru.otus.services.PlayerService",
+                "EquationPreparer, ru.otus.services.EquationPreparer",
+                "EquationPreparerImpl, ru.otus.services.EquationPreparer",
+                "equationPreparer, ru.otus.services.EquationPreparer"
+            })
+    void shouldExtractFromContextCorrectComponentWithNotNullFields(String classNameOrBeanId, Class<?> rootClass)
+            throws Exception {
+        var ctx = new AppComponentsContainerImpl(AppConfig1.class);
```

Comment:
```suggestion
        var ctx = new AppComponentsContainerImpl(AppConfig1.class, AppConfig2.class);
```
Если разбили конфигурацию, то надо передать оба класса конфигурации.